dev/setup/phpunit/setup_conf.sh

```diff
@@ -110,8 +126,20 @@ if [ "$DB" = 'mysql' ] || [ "$DB" = 'mariadb' ] || [ "$DB" = 'postgresql' ]; the
 	echo "MySQL flush"
 	${SUDO} "${MYSQL}" -u "$DB_ROOT" -h 127.0.0.1 $PASS_OPT -e 'FLUSH PRIVILEGES;'
 
-	echo "MySQL load sql"
-	${SUDO} "${MYSQL}" --force -u "$DB_ROOT" -h 127.0.0.1 $PASS_OPT -D travis < ${TRAVIS_BUILD_DIR}/dev/initdemo/mysqldump_dolibarr_3.5.0.sql | tee $TRAVIS_BUILD_DIR/initial_350.log
+	sum=$(find "${TRAVIS_BUILD_DIR}/htdocs/install" -type f -exec md5sum {} + | LC_ALL=C sort | md5sum)
+	load_cache=0
+	if [ -r "$DB_CACHE_FILE".md5 ] && [ -r "$DB_CACHE_FILE" ] && [ -x "$(which "${MYSQLDUMP}")" ] ; then
+		cache_sum="$(<"$DB_CACHE_FILE".md5)"
+		[ "$sum" = "$cache_sum" ] && load_cache=1
+	fi
+
+	if [ "$load_cache" = "1" ] ; then
+		echo "MySQL load cached sql"
+		${SUDO} "${MYSQL}" --force -u "$DB_ROOT" -h 127.0.0.1 $PASS_OPT -D travis < ${DB_CACHE_FILE} | tee $TRAVIS_BUILD_DIR/db_from_cacheinit.log
```

Comment:
The script computes an MD5 value on files in install.
Only when the MD5 value did not change the process is not repeated.
When the MD5 value changes, the process is repeated.
Possibly some files need to be added to make it cover more cases (if the init demo file is changed as well form instance)

As this is currently run in the Windows CI workflow which did not exist before, there is no impact on the other CI (travis).
Other conditions could be added to trigger the validation of the migration flow regularly:
- Change the cache-key to have it change once a week for instance or once every day.
- To make sure to have a fresh cache for the other jobs, run the CI on develop at midnight;
- Change the cache key when the workflow is manually triggered;
- Review the files that are included in the MD5 key.

Applying the migration on every CI run is long (can be more than 2 minutes), so it's better cached on most occasions.

As long as Travis CI is not updated, there is no regression versus before.

At this time there are still difficulties to recover the cache - the reason is not clear - the cache is saved but on restart the flow says no cache can be found - maybe because I force-push (I did not check in detail).
Locally recovering the cached file mechanism works.